src/test/java/org/apache/xml/security/encryption/keys/content/derivedKey/HKDFTest.java

```diff
@@ -0,0 +1,144 @@
+                    "000102030405060708090a0b0c," +
+                    "f0f1f2f3f4f5f6f7f8f9," +
+                    "42, " +
+                    "9b6c18c432a7bf8f0e71c8eb88f4b30baa2ba243," +
+                    "085a01ea1b10f36933068b56efa5ad81a4f14b822f5b091568a9cdd4f155fda2c22e422478d305f3f896",
+            "'Rfc5869: Test Case 5','http://www.w3.org/2000/09/xmldsig#sha1', " +
+                    "000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f303132333435363738393a3b3c3d3e3f404142434445464748494a4b4c4d4e4f," +
+                    "606162636465666768696a6b6c6d6e6f707172737475767778797a7b7c7d7e7f808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9fa0a1a2a3a4a5a6a7a8a9aaabacadaeaf," +
+                    "b0b1b2b3b4b5b6b7b8b9babbbcbdbebfc0c1c2c3c4c5c6c7c8c9cacbcccdcecfd0d1d2d3d4d5d6d7d8d9dadbdcdddedfe0e1e2e3e4e5e6e7e8e9eaebecedeeeff0f1f2f3f4f5f6f7f8f9fafbfcfdfeff," +
+                    "82, " +
+                    "8adae09a2a307059478d309b26c4115a224cfaf6," +
+                    "0bd770a74d1160f7c9f12cd5912a06ebff6adcae899d92191fe4305673ba2ffe8fa3f1a4e5ad79f3f334b3b202b2173c486ea37ce3d397ed034c7f9dfeb15c5e927336d0441f4c4300e2cff0d0900b52d3b4",
+            "'Rfc5869: Test Case 6','http://www.w3.org/2000/09/xmldsig#sha1', " +
+                    "0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b," +
+                    "''," +
+                    "''," +
+                    "42, " +
+                    "da8c8a73c7fa77288ec6f5e7c297786aa0d32d01," +
+                    "0ac1af7002b3d761d1e55298da9d0506b9ae52057220a306e07b6b87e8df21d0ea00033de03984d34918",
+            "'Rfc5869: Test Case 7','http://www.w3.org/2000/09/xmldsig#sha1', " +
+                    "0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c," +
+                    "," +
+                    "''," +
+                    "42, " +
+                    "2adccada18779e7c2077ad2eb19d3f3e731385dd," +
+                    "2c91117204d745f3500d636a62f64f0ab3bae548aa53d423b0d1f27ebba6f5e5673a081d70cce7acfc48",
+    })
+    void deriveKey(String name, String hash, String ikm, String salt, String info, int length, String prk, String okm) throws XMLSecurityException {
+        byte[] saltBytes = hexStringToByteArray(salt);
+        byte[] infoBytes = hexStringToByteArray(info);
+        byte[] ikmBytes = hexStringToByteArray(ikm);
+
+        byte[] expectedPRK =hexStringToByteArray(prk);
+        byte[] expectedOKM =hexStringToByteArray(okm);
+        String hMacHashAlgorithmURI = getHMacHashForHashUri(hash);
+        HKDF hkdf = new HKDF(hMacHashAlgorithmURI, saltBytes);
+        byte[] extractedKey = hkdf.extractKey(ikmBytes);
+        byte[] derivedKey = hkdf.deriveKey(ikmBytes, infoBytes, 0, length);
+
+        assertArrayEquals(expectedPRK, extractedKey);
+        assertArrayEquals(expectedOKM, derivedKey);
+    }
+
+
+    public static byte[] hexStringToByteArray(String value){
```

Comment:
Why can't we use the method in XMLCIpherUtil that was added instead of this?

Reply:
Good catch. Updated and reused the method for tests as well.